alephium/contracts/token_bridge/wrapped_alph.ral

```diff
@@ -0,0 +1,34 @@
+Contract WrappedAlph(
+    mut totalWrapped: U256
+) {
+    @using(preapprovedAssets=true, assetsInContract=true, permissionCheck=false)
+    pub fn wrapAndTransfer(
+        fromAddress: Address,
+        totalAmount: U256
+    ) -> () {
+        totalWrapped = totalWrapped + totalAmount
+        transferAlphToSelf!(fromAddress, totalAmount)
+        let self = selfAddress!()
+        WrappedAlphPool(callerContractId!()).receiveWrappedAlph{
+            self -> selfTokenId!(): totalAmount
+        }(self, totalAmount)
+    }
+
+    @using(preapprovedAssets=true, assetsInContract=true, permissionCheck=false)
+    pub fn unwrapAndCompleteTransfer(
+        fromAddress: Address,
+        recipient: Address,
+        transferAmount: U256,
+        arbiter: Address,
+        arbiterFee: U256
+    ) -> () {
+        let totalAmount = transferAmount + arbiterFee
```

Comment:
Do we want to check the caller of this method?

Reply:
Same as above.